test/test.js

```diff
@@ -649,7 +649,79 @@ tape('Socket gets re-opened as needed.', function (t) {
       logger.log(3, 'qwerty');
     }, 500);
   }, 500);
+});
+
+
+tape('Logger minLevel option is supported and works', function (t) {
+  t.plan(1);
+  t.timeoutAfter(1000);
+
+  const logger = new Logger({
+    token,
+    region: 'eu',
+    minLevel: 'info',
+  });
+
+  mockTest(_ => {
+    t.fail('Data should not be sent with lower logger level.');
+  });
+
+  logger.debug('asd');
+  logger.log('debug', 'asd');
+  logger['debug']('asd');
+
+  t.pass('Test finished');
+});
+
+tape('Winston JSON logger minLevel option is supported and works', function (t) {
+  t.plan(1);
+  t.timeoutAfter(1000);
+
+  const logger = winston.createLogger({
+    transports: [
+      new winston.transports.Insight({
+        token,
+        region: 'eu',
+        minLevel: 'info',
+        json: true,
+      }),
+    ]
+  });
+
+  mockTest(_ => {
+    t.fail('Data should not be sent with lower logger level.');
+  });
+
+  logger.debug('asd');
+  logger.log('debug', 'asd');
+  logger['debug']('asd');
+
+  t.pass('Test finished');
+});
+
+tape('Winston String logger minLevel option is supported and works', function (t) {
+  t.plan(1);
+  t.timeoutAfter(1000);
+
+  const logger = winston.createLogger({
+    transports: [
+      new winston.transports.Insight({
+        token,
+        region: 'eu',
+        minLevel: 'info',
+      }),
+    ]
+  });
+
+  mockTest(_ => {
+    t.fail('Data should not be sent with lower logger level.');
```

Comment:
this test would never actually fail as it stands now, right?

Reply:
nope, it will fail if I take away the implementation
```
  Failed Tests: There were 8 failures
    32) Winston JSON logger minLevel option is supported and works
      ✗ Data should not be sent with lower logger level.
      ✗ plan != count
      ✗ Data should not be sent with lower logger level.
      ✗ Data should not be sent with lower logger level.
    34) Socket is not closed after inactivity timeout when buffer is not empty.
      ✗ Data should not be sent with lower logger level.
      ✗ plan != count
      ✗ Data should not be sent with lower logger level.
      ✗ Data should not be sent with lower logger level.
```

Reply:
Why haven't you added `logger.log({level: 'debug', message:'msg' })?`
It also may be worth to add a test with custom log_ levels with symbols. it's quite common pattern that you may want to support as well